packages/jest-core/src/SearchSource.ts

```diff
@@ -299,6 +300,9 @@ export default class SearchSource {
       const tests = searchResult.tests;
 
       const filter = require(filterPath);
+      if (filterSetupPromise) {
+        await filterSetupPromise;
```

Comment:
Um, why do we pass the filter promise instead of taking it right from `filter` and just awaiting it? I get that it may be slightly faster (but really, is it actually measurable difference?) but it couples searchsource to cli too much (filter is actually required in 2 places, which is bad). 

For the record, we recently got rid of changedFilesPromise for similar reasons.

Reply:
The Promise was called earlier and passed down. It gets called before Haste Map generation, which takes several seconds. In FB's case, which is what I measured, if you have an HTTP call that takes longer than 200ms , anything on top of the 200ms is cut off. It's a measurable difference.

Any suggestions to avoid coupling?

Reply:
To clarify-- ALL time spent waiting on an HTTP request is cut off, because before it wasn't parallelized against anything, so it can be a significant difference.

The reason there's 200ms-ish of overlap is because, depending on the size of the response, the HTTP request still needs to be parsed, and the haste map generation can CPU block up to that point, so the response doesn't get parsed.

In my tests, an HTTP call of 1.2s~ had a solid second cut off due to the change.

Reply:
You're right, I didn't notice it's invoked before building haste map. I don't have any clever ideas around how to avoid coupling. At least it's inside the same module (`jest-core`).
Maybe we could just pass whole required filter module (with initialized `setup` call) and then await it when necessary? This would avoid requiring filter twice. If that seems cleaner, feel free to do it, but if not, I'm good with accepting this in current shape

Reply:
Sure, I can get behind that. If we're passing something anyway, might as well pass the whole filter with the setup bootstrap already done. Something like (example simplified):

```js
const setupPromise = filter.setup();
const filterToPass = (testPaths: Array<string) => {
    await setupPromise;
    return filter(testPaths);
};

...

// Later:
filterToPass(testPaths);
```

I think it's a good idea, working on it now. Thanks for the feedback.

Reply:
Yea, makes sense this way, thanks as well!